### ISSUE TITLE
(Late) Game Extensions and minor tweaks

### ISSUE THREAD

**Issue by WatKey** [Anuken/Mindustry-Suggestions#2982]
### Describe the content or mechanics you are proposing.

**Disclaimer:** I only recently started playing the game and have not yet finished the campaign. Therefore I am not very familiar with all the technologies and blocks in the game, however I did some research on the wiki and trello and its seems like the additions I am proposing here are not yet part of the game or in development. The names I have used here are only for descriptive purposes and can of course be replaced or improved. Newly introduced concepts should be in **bold** font. Please point out any mistake or inconsistency you find and feel free to add your own ideas and concepts to my proposals, as there probably is a lot more space to improve my suggestions.

- **Ferromagnets**: Used for constructing magnets that could be used in _Mass drivers_, _Advanced turrets_ (maybe **Rail Guns**?), _magnetic transport means_, **Particle Accelerators** and **Antimatter** storage.

- **Electromagnets** (advanced tech required): Stronger than Ferromagnets, requires copper and Ferromagnets to build. Uses large amounts of electricity, however increases damage of advanced turrets (if used to build them), allows for higher energies in the particle accelerator, faster transport means (maybe **magnetic repulser tech**, that would increase the speed of units?)

- Option for choosing which magnetic material to use for a block to be build, resulting in an easy tiering system of blocks that use magnetic materials.

- Electromagnets may be turned into **Superconductors** by cooling them. This would greatly increase the intensity of the created magnetic fields, while reducing the power consumption because less energy is lost in the form of heat. 

- **Cryofluid Tiers:** So far Water and Cryofluid exist, however it could be beneficial to split Cryofluid into different tiers, such as liquid nitrogen and liquid helium, allowing for different tiers of superconductors. Liquid helium for example could be generated by using a **Fusion Reactor**, fusing hydrogen cores into helium cores, whereas liquid nitrogen could simply take the place of the current Cryofluid. I am using the real names here, however one could of course keep the name Cryofluid and instead additionally introduce something called **Hypocryofluid**, for example.

- **Fusion Reactor**: Requires large amounts of energy to be started, then however produces a lot more energy than is consumed (similar to the Impact Reactor). Consumes water and produces heavier nuclear cores in the process that can be used for advanced cooling technologies (see above). The reactor requires magnetic materials and the more advanced they are, the more energy and products are produced.

- **Particle Accelerator:** Large Multi Block Structure that allows to unlock advanced technology such as **Nuclear** (seems to be in development) and **Quantum Physics** (**Quantum Processor/Computer** maybe?), **Wormhole Theory** and **Antimatter Studies**. The accelerator itself could be composed of multiple accelerator tubes together with detector blocks that allow for the detection of the particles. The larger the accelerator and the more energy is available for the experiments and the better the cooling, the higher the tier of technologies that can be unlocked. Moreover allows for the creation of **Antimatter**. Could also be used as the block that allows to enrich and alter nuclear materials (maybe interesting for the ongoing development of nuclear techs). Also an interesting synthesis with Processors as they would be able to automatise the experiments and the production at the accelerator. The accelerator requires large amounts of energy and the processor could therefore be programmed to make sure the accelerator is only running when a certain energy threshold is reached, therefore making sure the accelerator does not take energy away from turrets that require it to defend against incoming waves.

- **Antimatter:** Has to be stored and transported within strong magnetic fields to avoid contact with matter. Can be used as ammunition (annihilates matter at contact) or as component in **Annihilation Reactors**.

- **Annihilation Reactor:** Uses Antimatter to annihilate any kind of matter and injects the resulting energy into the power grid.

- **Wormholes:** End Game technology that allows for the creation of either temporary (“teleport” your units towards the enemy) or lasting (“teleport” materials from one wormhole to one next to the core) Wormholes that require _ludicrous_ amounts of energy. 

- **Wormhole Projector:** Block that steers the creation and placement of wormholes.

### Describe how you think this content will improve the game. If you're proposing new content, mention how it may add more gameplay options or how it will fill a new niche.

The above mentioned additions and changes introduce the **Particle Accelerator** as a new core concept to unlock additional technologies and to produce late game materials. This structure is a large multi block structure and introduces the concept of structures that inherently come with different upgrade tiers, as the size, the amount of energy, the magnetic materials used and the cooling system can be varied. The other changes are mostly prerequisites or logical consequences to the introduction of the particle accelerator. Although there already are quite a lot of materials in the game, I believe it would make sense to introduce the new resource of **Ferromagnetic Materials** (**Ferromagnets**), as they not only become important for the aforementioned accelerator, but could also be incorporated to be required by already existing blocks, such as the mass driver for example. Magnets are interesting structures and it therefore would render the game even more interesting by having the possibility to use different magnet tiers through **Electromagnets** and **Supermagnets** (**Superconductors**). Introducing different cryofluid tiers would further increase the different combinations of tiers, however this change is not strictly necessary. The introduction of wormhole technologies would bring a new late game component that one may try to chase as the projection of wormholes should be so expensive that only gigantic and incredibly advanced industrial complexes should be able to produce the required energy.

### Before making this issue, check the boxes below to confirm that you have acknowledged them.

- [X] I have checked the [Trello](https://trello.com/b/aE2tcUwF/mindustry-trello) to make sure my suggestion isn't planned or implemented in a development version.
- [X] I am familiar with all the content already in the game or have glanced at the wiki to make sure my suggestion doesn't exist in the game yet.
- [X] I have read [README.md](https://github.com/Anuken/Mindustry-Suggestions/blob/master/README.md) to make sure my idea is not listed under the "A few things you shouldn't suggest" category.

**Comment by OnyxFinger** [comment]
cryofluid tiers isn't an idea I'm in : the game is mainly played by the community because it's simple (kind of a shame but idc cuz PvP is still somehow good with how bad it also is so I play it), so I think adding multiple tiers of the same resource can be confusing and too complicated for the player base

the fusion reactor is just an impact reactor : it makes a lot of power but requires a lot to kickstart it. I think currently, an impact reactor's insane power production is enough for the game, but since new planets and stuff are coming wiht different tech etc... you get it, this idea may be useful still.

about wormholes, I honestly just don't want to brainstorm about it, so I'm not gonna think about it sorry

For the particle accelerator, it should be a factory, since that's pretty much how unlocking technology works in this game (e.g. getting plastanaium allows you to place overdrive projectors, cyclones and swarmers

antimatter : surge is already an endgame ammo, and tbh endgame really doesn't need a buff rn. About the antimatter reactor, see above with fusion reactors

**Comment by oinkymoo** [comment]
magnets: there are at least 2 railguns already in the game. neither of them use magnets.

cryofluid: you do know that cryofluid is currently created using water and titanium, right? calling that stuff liquid nitrogen wouldn't make much sense

generators: i see no purpose to this. this would only be necessary if something required orders of magnitude more power than an impact reactor provides

**Comment by IVV-NCX** [comment]
I played a mod with much of these items, called [NewHorizonsMod](https://github.com/Yuria-Shikibe/NewHorizonMod), and contains your fusion reactor, teleporters, ultrapowerful weapons, adds some changes to early game, and more.

The above is all right. I do agree that unit teleporters or high speed transport would be cool, although that can be done with logic (I'm not sure, I don't mindustry much these days).

**Comment by OnyxFinger** [comment]
actually wormholes are for units aren't they ?
I think if it's restricted to units they're fine

**Comment by IVV-NCX** [comment]
Yeah, they are. But if you think about it, isn't the unit transport already there with the logic units? Plus, doesn't it break the canonical amount of technology? I'm 99% sure that the tech only extends to nuclear and basic AI (about our level).
